filebeat/input/v2/compat/compat.go

```diff
@@ -176,3 +184,24 @@ func configID(config *conf.C) (string, error) {
 
 	return fmt.Sprintf("%16X", id), nil
 }
+
+func (f *factory) generateCheckConfig(config *conf.C) (*conf.C, error) {
+	testCfg, err := conf.NewConfigFrom(config)
+	if err != nil {
+		return config, fmt.Errorf("failed to create new config: %w", err)
+	}
+
+	// let's try to override the `id` field, if it fails, give up
+	inputID, err := testCfg.String("id", -1)
+	if err != nil {
+		return config, fmt.Errorf("failed to get 'inputID': %w", err)
+	}
+
+	// using math/rand for performance, generate a 0-9 string
+	err = testCfg.SetString("id", -1, inputID+strconv.Itoa(rand.Intn(10)))
```

Comment:
You can use math/rand/v2 `rand.IntN(10)` here, which is even faster. There was some effort in the past to migrate math/rand to math/rand/v2, another reason to stick with v2 in new code.